gitprovider/types_repository.go

```diff
@@ -58,6 +58,12 @@ type RepositoryInfo struct {
 	// Default value at POST-time: RepositoryVisibilityPrivate.
 	// +optional
 	Visibility *RepositoryVisibility `json:"visibility"`
+
+	// Name returns the name of the repository.
+	// No default value at POST-time.
+	// Added for Azure devops compatibility since it allows updating the name
+	// +optional
+	Name *string `json:"name"`
```

Comment:
This type is also returned by e.g. `UserRepository.Get`. We should adapt the other implementations to properly fill this field. Otherwise it will be nil which feels strange, esp. since the information is available from the API types.